Possible flaky test log anonymization

## Problem Definition

Unexpected failure happened here: https://github.com/rotki/rotki/pull/944/checks?check_run_id=625213750

It was an OSX github action run

```
2020-04-28T08:01:50.0056988Z =================================== FAILURES ===================================
2020-04-28T08:01:50.0063744Z _________________________ test_log_anonymization[True] _________________________
2020-04-28T08:01:50.0064287Z 
2020-04-28T08:01:50.0064592Z anonymized_logs = True
2020-04-28T08:01:50.0064893Z caplog = <_pytest.logging.LogCaptureFixture object at 0x7fa4681b5e90>
2020-04-28T08:01:50.0065113Z 
2020-04-28T08:01:50.0066068Z     @pytest.mark.parametrize('anonymized_logs', [True, False])
2020-04-28T08:01:50.0066393Z     def test_log_anonymization(anonymized_logs, caplog):
2020-04-28T08:01:50.0066709Z         """Tests the implementation of sensitive log anonymization"""
2020-04-28T08:01:50.0067002Z         caplog.set_level(logging.INFO)
2020-04-28T08:01:50.0067288Z         LoggingSettings(anonymized_logs=anonymized_logs)
2020-04-28T08:01:50.0067568Z         log_keys = {
2020-04-28T08:01:50.0068049Z             'amount': 55,
2020-04-28T08:01:50.0069668Z             'cost': FVal('10.5'),
2020-04-28T08:01:50.0070488Z             'gas': 33333333333333,
2020-04-28T08:01:50.0071024Z             'fee': FVal('0.0001'),
2020-04-28T08:01:50.0072066Z             'timestamp': 1577003884,
2020-04-28T08:01:50.0072601Z             'eth_address': '0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5',
2020-04-28T08:01:50.0075057Z             'eth_accounts': [
2020-04-28T08:01:50.0075700Z                 '0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5',
2020-04-28T08:01:50.0076224Z                 '0xfeb0670822d4c9c8ae7b02dc7bf095251ce42ced',
2020-04-28T08:01:50.0076540Z             ],
2020-04-28T08:01:50.0077132Z             'tx_hash': '0x16dac02d1e3597eed3b6bc645b9d7c59554bb07c2a90219af1b761924d5ef7db',
2020-04-28T08:01:50.0077453Z         }
2020-04-28T08:01:50.0077734Z         log.info(
2020-04-28T08:01:50.0078177Z             'Testing anonymization',
2020-04-28T08:01:50.0078480Z             sensitive_log=True,
2020-04-28T08:01:50.0078764Z             **log_keys,
2020-04-28T08:01:50.0079037Z         )
2020-04-28T08:01:50.0082767Z     
2020-04-28T08:01:50.0083253Z         assert 'sensitive_log' not in caplog.text, 'The sensitive log attribute should be deleted'
2020-04-28T08:01:50.0083405Z     
2020-04-28T08:01:50.0083536Z         for key, value in log_keys.items():
2020-04-28T08:01:50.0083828Z             entry = f'{key}={str(value)}'
2020-04-28T08:01:50.0083962Z             if anonymized_logs:
2020-04-28T08:01:50.0084248Z                 assert key + '=' in caplog.text
2020-04-28T08:01:50.0084571Z >               assert entry not in caplog.text, f'{key} entry should have been modified'
2020-04-28T08:01:50.0084722Z E               AssertionError: amount entry should have been modified
2020-04-28T08:01:50.0085039Z E               assert 'amount=55' not in 'INFO     ro...3b55763b55\n'
2020-04-28T08:01:50.0085328Z E                 'amount=55' is contained here:
2020-04-28T08:01:50.0085970Z E                   ymization amount=5597.414, cost=7778.535, gas=38003390, fee=4.547, timestamp=1529626313, eth_address=0x0c3d7c2f487020724c77406a5a2b735b2b5a274f, eth_accounts=['0x7931403343236370586841263c5b0c430c46547a', '0x7931403343236370586841263c5b0c430c46547a'], tx_hash=0x684d730d2b54587031544140734b45412e5a46336c4d46296f29453b55763b55
2020-04-28T08:01:50.0086113Z 
2020-04-28T08:01:50.0086244Z rotkehlchen/tests/unit/test_logging.py:42: AssertionError
2020-04-28T08:01:50.0086520Z travis_fold:start:py-6219.log.0
2020-04-28T08:01:50.0086837Z ------------------------------ Captured log call -------------------------------
2020-04-28T08:01:50.0087549Z INFO     rotkehlchen.tests.unit.test_logging:test_logging.py:33 Testing anonymization amount=5597.414, cost=7778.535, gas=38003390, fee=4.547, timestamp=1529626313, eth_address=0x0c3d7c2f487020724c77406a5a2b735b2b5a274f, eth_accounts=['0x7931403343236370586841263c5b0c430c46547a', '0x7931403343236370586841263c5b0c430c46547a'], tx_hash=0x684d730d2b54587031544140734b45412e5a46336c4d46296f29453b55763b55
2020-04-28T08:01:50.0087877Z travis_fold:end:py-6219.log.0
```

## Task

Check it out and try to reproduce flakiness locally

--- a/rotkehlchen/tests/fixtures/accounting.py
+++ b/rotkehlchen/tests/fixtures/accounting.py
@@ -83,7 +83,7 @@ def accountant(
         database,
         data_dir,
         accounting_create_csv,
-        messages_aggregator,
+        function_scope_messages_aggregator,
         start_with_logged_in_user,
         accounting_initialize_parameters,
 ) -> Optional[Accountant]:
@@ -93,7 +93,7 @@ def accountant(
     accountant = Accountant(
         db=database,
         user_directory=data_dir,
-        msg_aggregator=messages_aggregator,
+        msg_aggregator=function_scope_messages_aggregator,
         create_csv=accounting_create_csv,
     )
 

--- a/rotkehlchen/tests/unit/test_logging.py
+++ b/rotkehlchen/tests/unit/test_logging.py
@@ -39,6 +39,8 @@ def test_log_anonymization(anonymized_logs, caplog):
         entry = f'{key}={str(value)}'
         if anonymized_logs:
             assert key + '=' in caplog.text
-            assert entry not in caplog.text, f'{key} entry should have been modified'
+            msg = f'{key} entry should have been modified'
+            assert entry not in caplog.text or entry + ',' not in caplog.text, msg
         else:
-            assert entry in caplog.text, f'{key} entry should not have been modified'
+            msg = f'{key} entry should not have been modified'
+            assert entry in caplog.text or entry + ',' in caplog.text

